fe/src/project/AspNetCore/areas/News/components/NewsCatalog/NewsCatalog.js

```diff
@@ -0,0 +1,156 @@
+import ResponsiveImage from "../../../../../../shared/components/Picture/ResponsiveImage";
+import QueryStringSerializer from "../../../../../../shared/helpers/serializers/QueryStringSerializer";
+import NavigationHelper from "../../../../../../shared/helpers/globals/NavigationHelper";
+import { Context } from "../../../../../../shared/stores/Store";
+import { CalendarToday } from "@material-ui/icons";
+import { Hash } from "react-feather"
+import moment from "moment";
+
+const NewsCatalog = (props) => {
+    const [state, dispatch] = useContext(Context);
+    const [items, setItems] = useState(props.pagedResults.data ? props.pagedResults.data : null);
+    const [total] = useState(props.pagedResults.total ? props.pagedResults.total : null)
+    const [hasMore, setHasMore] = useState(true);
+
+    let pageIndex = 2;
+    const handleLoadNews = async () => {
+        if (total < 10) return;
+
+        dispatch({ type: "SET_IS_LOADING", payload: true });
+        const requestOptions = {
+            method: "GET",
+            headers: { "Content-Type": "application/json", "X-Requested-With": "XMLHttpRequest" }
+        }
+
+        const queryStrings = {
+            itemsPerPage: 9, 
+            pageIndex: pageIndex
+        }  
+
+        const getNewsUrl = props.newsApiUrl + "?" + QueryStringSerializer.serialize(queryStrings);
+        await fetch(getNewsUrl, requestOptions)
+            .then((response) => {
+                dispatch({ type: "SET_IS_LOADING", payload: false });
+
+                return response.json().then(jsonResponse => {
+                    if (response.ok){
+                        jsonResponse.data.forEach(item => {
+                            setNews(element => [...element, item]);
+                        })
+
+                        pageIndex += 1;
+
+                        if (pageIndex > jsonResponse.pageCount){
+                            setHasMore(false)
+                        }
+                    }
+                });
+            }).catch(() => {
+                setHasMore(false)
+            });
+    }
+
+    const handleDynamicScroll = (e) => {
+        const scrollHeight = e.target.documentElement.scrollHeight;
+        const currentHeight = Math.ceil(
+            e.target.documentElement.scrollTop + window.innerHeight
+        );
+
+        if (currentHeight + 1 >= scrollHeight) {
+            handleLoadNews();
+        }
+    }
+
+    useEffect(() => {
+        window.addEventListener("scroll", handleDynamicScroll)
+
+        return () => window.removeEventListener("scroll", handleDynamicScroll)
+    }, [hasMore])
+
+    const handleCategory = (category) => {
+        if (!category){
+            return setItems(props.pagedResults.data);
+        }
+
+        const filtered = props.pagedResults.data.filter((item) => item.categoryName === category.name);
+
+        return setItems(filtered);
+    }
+
+    const navigateToNews = (item) => {
```

Comment:
**suggestion (non-blocker):** use NavigationHelper.redirect(item.url) directly, there is no need to create navigateToNews wrapper for this one liner.